core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1142,6 +1142,19 @@ class KafkaApis(val requestChannel: RequestChannel,
     val metadataRequest = request.body[MetadataRequest]
     val requestVersion = request.header.apiVersion
 
+    // Topic IDs are not supported for versions 10 and 11. Topic names can not be null in these versions.
+    if (metadataRequest.version() >= 10 && !metadataRequest.isAllTopics) {
+      metadataRequest.data().topics().forEach{ topic =>
+        // If null, set to the empty string, since the response does not allow null.
+        if (topic.name() == null) {
+          topic.setName("")
+          throw new InvalidRequestException(s"Topic name can not be null for version ${metadataRequest.version()}")
+        } else if (topic.topicId() != Uuid.ZERO_UUID) {
+          throw new InvalidRequestException(s"Topic IDs are not supported in requests for version ${metadataRequest.version()}")
+        }
+      }
+    }
```

Comment:
Can we not do such validation in the request class?

Reply:
We can do some there. But would we still need this since the protocol also allows for nulls/topic Ids?

Reply:
What I mean is that this logic could exist in the request class and you can then call the method from here. That way it's much easier to test.